docs: Add warning about saving credentials (#5726)
python-poetry/poetry: docs/repositories.md

Then, assuming the repository requires authentication, configure credentials for it.

```bash
poetry config http-basic.foo <username> <password>
```

{{% warning %}}

> Depending on your systems configuration, credentials might be saved in your command lines history.

Author: ```suggestion
Depending on your system's configuration, credentials might be saved in your command line history.
```

> Depending on your system's configuration, credentials might be saved in your command line history.

Reviewer: Could be worth mentioning that many shells do not save space-prefixed commands into history. It's not universal, but it's a very common feature for this use case.

> Depending on your system's configuration, credentials might be saved in your command line history.

Author: ```suggestion
Depending on your system configuration, credentials might be saved in your command line history.
Many shells avoid saving command to history if it is prefixed by a space character. For more information, please refer to your shells manual.
```

Then, assuming the repository requires authentication, configure credentials for it.

```bash
poetry config http-basic.foo <username> <password>
```

{{% warning %}}
Depending on your system configuration, credentials might be saved in your command line history.

> Many shells avoid saving command to history if it is prefixed by a space character. For more information, please refer to your shells manual.

Reviewer: ```suggestion
Many shells do not save commands to history when they are prefixed by a space character. For more information, please refer to your shell's documentation.
```
